Init database first snapshot is ready to be migrated

EbrahimMo50 opened the pull request:
migration file is set, database is ready to be updated

EbrahimMo50 committed:
Init database first snapshot is ready to be migrated

MahmoudMNael commented on AuraChat/Entities/AppDbContext.cs:
```diff
@@ -4,8 +4,35 @@ namespace AuraChat.Entities;
+
+        modelBuilder.Entity<Message>(entity =>
+        {
+            entity.OwnsMany(m => m.RecieverState);
```
Why does that ownsMany?? is is because when the message is sent in a group? so it would hold receiver states for all group members?

MahmoudMNael commented on AuraChat/Entities/Group.cs:
```diff
@@ -2,14 +2,13 @@
+    public int Id { get; private set; }
     public string Name { get; set; } = null!;
     public string? Description { get; set; }
     public string? GroupPictureGUID { get; set; }
```
change name here to follow the agreed upon convention

MahmoudMNael commented on AuraChat/Entities/Group.cs:
```diff
@@ -2,14 +2,13 @@
+    public ICollection<Message> Messages { get; set; } = [];
     public GroupSettings Settings { get; set; } = null!;
     public int CreaterId { get; set; }
     public User Creater { get; set; } = null!;
```
its called Creator

MahmoudMNael commented on AuraChat/Entities/Message.cs:
```diff
@@ -2,12 +2,12 @@
+    public ICollection<RecieverState> RecieverState { get; set; } = [];
     public DateTime SentTime { get; set; } = DateTime.UtcNow;
     public string Content { get; set; } = null!;
     public string? MediaGUID { get; set; }
```
same here... follow naming convention

MahmoudMNael commented on AuraChat/Entities/User.cs:
```diff
@@ -1,16 +1,19 @@
     public string Salt { get; set; } = null!;
     public string HashedPassword { get; set; } = null!;
     public string? Bio {  get; set; }
     public string? ProfilePictureGUID { get; set; }
```
naming convention

MahmoudMNael commented on AuraChat/Entities/User.cs:
```diff
@@ -1,16 +1,19 @@
+    public ICollection<Chat> Chats { get; set; } = [];
+    public ICollection<UserGroup> UsersGroups { get; set; } = [];
+    public ICollection<Group> Groups { get; set; } = [];
+    public ICollection<User> Friends { get; set; } = [];
```
I'd like to discuss more about the friends feature how would it be implemented? as i see nothing in the dbcontext specifing a many:many relation here

Or efcore by default implement it as many to many?

from what i know when there is an array in a entity and not specified the relation its by default a 1 to many not many to many

MahmoudMNael replied:
so maybe clarify more on this part

EbrahimMo50 committed:
Fix applied the latest reviews on the branch

adjusted properties names and re-applied the migration to change the name stored in the Db table
